pkg/fileutils/fileutils.go

```diff
@@ -98,7 +98,22 @@ func (pm *PatternMatcher) Matches(file string) (matches uint, err error) {
 		logrus.Debugf("Skipping excluded path: %s", file)
 	}
 
-	return matches, nil
+	return matches, excludes, nil
+}
+
+// IsMatch verifies the provided filepath against all patterns and returns true
+// if it overall matches. It returns an error on failure and is not safe to be
+// called concurrently.
+func (pm *PatternMatcher) IsMatch(file string) (matched bool, err error) {
+	matches, excludes, err := pm.Matches(file)
+	if err != nil {
+		return false, err
+	}
+
+	if matches >= excludes && matches-excludes > 0 {
```

Comment:
The two conditions seem to contradict a bit (one allows equality while the other doesn't) and are redundant. If we expect more matches than exclusions, `matches-excludes > 0` is sufficient.
```golang
    ...
    return matches-excludes > 0, nil
}
```

Reply:
I see you point and will remove the `=`, but it will overflow if `excludes > matches` because we're dealing with unsigned integers here.

Reply:
Eeeeek, good catch. Didn't think about the underflow :+1: